libsignal-protocol/src/keys/private.rs

```diff
@@ -0,0 +1,79 @@
+use crate::{
+    errors::FromInternalErrorCode, keys::PublicKey, raw_ptr::Raw, Context,
+};
+use failure::Error;
+use std::{
+    cmp::{Ord, Ordering},
+    ptr,
+};
+
+pub struct PrivateKey {
+    pub(crate) raw: Raw<sys::ec_private_key>,
+}
+
+impl PrivateKey {
+    pub fn decode_point(
+        ctx: &Context,
+        data: &[u8],
```

Comment:
It would be nice to name that arg `key` maybe?

As `data` sounds more generic.